File: https:/github.com/Karuna212006/Leet_code/53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 7/14/2026, 2:01:07 PM
class Solution {
    public int maxSubArray(int[] arr) {
        int max=arr[0];
        int submax=0;
        for(int a:arr){
            submax+=a;
            max=Math.max(max,submax);
            if(submax <0)submax=0;
        }
        return max;
    }
}

File: https:/github.com/Karuna212006/Leet_code/53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 7/14/2026, 2:01:01 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        m=nums[0]
        c=0
        for i in nums:
            if(c<0):
                c=0
            c+=i
            if(c>m):
                m=c
        return m